### ISSUE TITLE
Job version constantly changed.

### ISSUE THREAD

**Issue by ont** [hashicorp/nomad#6416]
### Nomad version
Output from `nomad version`
```
Nomad v0.9.4 (a81aa846a45fb8248551b12616287cb57c418cd6)
```
and
```
Nomad v0.9.3 (c5e8b66c3789e4e7f9a83b4e188e9a937eea43ce)
```
for second datacenter.

### Operating system and Environment details

- two datacenters with `Ubuntu 18.04.2 LTS` and `Ubuntu 18.04.3 LTS`
- first datacenter configured to use only external ip addresses.
- second datacenter configured to use external ip addresses and private ip from 10.0.0.0/24 subnet.
- two datacenter interconnected via consul via `-advertise-wan` and `-retry-join-wan` consul's options.
- some random errors in nomad debug log about connection to other members.

### Issue
Output from `nomad status <allocation-id>` shows **constantly changing** `Job Version`. 
For example:
```
# nomad status 54dfb888

ID                  = 54dfb888
Eval ID             = 59ea4550
Name                = caddy.caddy[0]
Node ID             = 405742e7
Node Name           = balancer
Job ID              = caddy
Job Version         = 824634494400
Client Status       = running
Client Description  = Tasks are running
Desired Status      = run
Desired Description = <none>
Created             = 1d3h ago
Modified            = 1d3h ago

....
```
After some short amount of time (1-2 seconds):
```
ID                  = 54dfb888
Eval ID             = 59ea4550
Name                = caddy.caddy[0]
Node ID             = 405742e7
Node Name           = balancer
Job ID              = caddy
Job Version         = 824635730528
Client Status       = running
Client Description  = Tasks are running
Desired Status      = run
Desired Description = <none>
Created             = 1d3h ago
Modified            = 1d3h ago
```
But output from `nomad inspect caddy | jq .Job.Version` is stable:
```
# nomad inspect caddy | jq .Job.Version
41
```

Does it normal behavior and what this big random numbers means?
This output is for any valid allocation id.

### Probably wrong code
`command/alloc_status.go`:
```
		fmt.Sprintf("Job ID|%s", alloc.JobID),
		fmt.Sprintf("Job Version|%d", *alloc.Job.Version),
		fmt.Sprintf("Client Status|%s", alloc.ClientStatus),
```
I don't understand why dereference `*alloc.Job.Version` is used here...

### Reproduction steps
Currently unknown.

### Nomad Server configs
Each datacenter has three server nodes with this configs
```
server {
    enabled = true
    bootstrap_expect = 3
    heartbeat_grace = "300s"
    min_heartbeat_ttl = "30s"
    max_heartbeats_per_second = 10.0
}
```
This nodes also works as clients.

### Nomad Server logs
Seems to be nothing interesting here:
https://gist.github.com/ont/b0fa29122f550dfc5f1827ae4aee954d

### Nomad Client logs
Seems to be nothing interesting here:
https://gist.github.com/ont/da9ba9c1642077a9474d43f627a380f8

**Comment by preetapan** [comment]
This was fixed in #6328

**Comment by preetapan** [comment]
Pic
